index.js

```diff
@@ -1,11 +1,18 @@
 module.exports = function (string, context) {
-  var key, find, re;
+  return _recursive_rendering(string, context, '');
+};
 
-  for (key in context) {
-    find = '\\$\\{\\s*' + key + '\\s*\\}';
-    re = new RegExp(find, 'g');
-    string = string.replace(re, context[key]);
+function _recursive_rendering(string, context, stack) {
+  for (var key in context) {
+    if (context.hasOwnProperty(key)) {
+      if(typeof context[key] === "object") {
+        string = _recursive_rendering(string, context[key], (stack?`${stack}.`:'')+ key);
```

Comment:
template literals are not compatible with ES5. I think we can keep this library simple and avoid use babel to transpile code to do it ES5 compatible.